ast-nodes/variable-declaration.dart

```diff
@@ -14,15 +16,32 @@ class VariableDeclaration extends Declaration {
   VariableDeclaration(name, this.type, this.value) : super(name);
 
   factory VariableDeclaration.parse(Iterable<Token> tokens) {
-    // TODO: write the actual parser body
-    return VariableDeclaration('dummy', null, Expression.parse(tokens));
+    var iter = tokens.iterator;
+    var temp_name;
+    iter.moveNext(); // Skip the `var` keyword
+    if (iter.moveNext() &&
+        iter?.current?.value != ':' &&
+        iter?.current?.value != 'is')
+      temp_name = iter.current.value;
+    else
+      throw SyntaxError(null, 'Expected a name of the variable');
+    if (!iter.moveNext())
+      throw SyntaxError(null, 'Type is required');
+    else if (iter.current.value == ':') {
```

Comment:
You're missing the case when both the type and the initial value are specified, like `var a: integer is 5`